File: ex1-palindromo/Palindromo.java
import java.util.*;

public class Palindromo {
  public static boolean compareStrings(String palavra1, String palavra2) {
    if(palavra1.length() != palavra2.length()) {
      return false;
    }

    for (int i = 0; i < palavra1.length() - 1; i++) {
      if(palavra1.charAt(i) != palavra2.charAt(i)) {
        return false;
      }
    }

    return true;
  };

  public static boolean checarPalavra(String palavra) {
    for (int i = 0; i < palavra.length() / 2; i++) {
      if(palavra.charAt(i) != palavra.charAt(palavra.length() - i - 1)) {
        return false;
      }
    }

    return true;
  };

  public static void main(String[] args) {
    MyIO.setCharset("UTF-8");
    String palavra;
    boolean ePalindromo;
    palavra = MyIO.readLine();

    while (!compareStrings(palavra, "FIM")) {
      ePalindromo = checarPalavra(palavra);

      if (ePalindromo) {
        System.out.println("SIM");
      } else {
        System.out.println("NAO");
      }

      palavra = MyIO.readLine();
    }
  };
}

File: ex2-palindromo-c/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int ePalindromo(char palavra[]) {
  int conta = 0, i = 0;
  while(palavra[i] != '\0') {
    conta++;
    i++;
  }

  i = 0;

  for(int i = 0; i < conta / 2; i++) {
    if (palavra[i] != palavra[conta - i - 1]) {
      return 0;
    }
  }

  return 1;
}

int main(void) {
  char palavra[300];
  scanf(" %[^\n]", palavra);

  while(strcmp(palavra, "FIM") != 0) {
    if(ePalindromo(palavra)) {
      printf("SIM\n");
    } else {
      printf("NAO\n");
    }
    getchar();
    scanf(" %[^\n]", palavra);
  }
  return 0;
}
